compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -507,7 +507,7 @@ OMR::ARM64::TreeEvaluator::loadaddrEvaluator(TR::Node *node, TR::CodeGenerator *
          }
       else
          {
-         TR_UNIMPLEMENTED();
+         generateTrg1MemInstruction(cg, TR::InstOpCode::addx, node, resultReg, mref);
```

Comment:
Which path in the binary encoding of the memory reference do you expect this instruction to take?  Since it is not an indexed form and not an `addimmx` my reading of the code finds it hits the assert on L882.  Can you clarify?

Reply:
Code generation of that path is handled by `J9MemoryReference.cpp` instead of `OMRMemoryReference.cpp` in https://github.com/eclipse/openj9/pull/6550 .
Should I override the function `loadaddrEvaluator()` in the `J9TreeEvaluator.cpp` in the same way as I did with MemoryReference classes?

Reply:
I think it would be cleaner moving the loadaddr implementation that deals with unresolved into OpenJ9, which is where the MemoryReference class handles it.  From an OMR perspective I think it would be more clear what's going on (because right now the code in OMR is creating a memory reference that can't be satisfied by the OMR implementation of MemoryReference).